### PR TITLE
Annotation tweaks

### PR DESCRIPTION
**Pull request by jzuhone**
This PR adds two tweaks to plot annotations.

* An offset for timestamps, e.g. if you want to make the timestamp shown on the plot to be (say) `ds.current_time` - 5 Gyr so that the zero is at a different time.

* A `format` argument for the `ScaleCallback` similar to the `time_format` argument for the `TimestampCallback`. 

Not sure what would be a good test here, still need to add notes to the narrative documentation. Suggestions welcome.

### CODE REVIEW

File: yt/visualization/plot_modifications.py [hunk 1]
```diff
@@ -1848,6 +1850,10 @@ class TimestampCallback(PlotCallback):
 
             "figure" -- the MPL figure coordinates: (0,0) is lower left, (1,1) is upper right
 
+    time_offset : float, (value, unit) tuple, or YTQuantity, optional
+        Apply an offset to the time shown in the annotation from the
+        value of the current time. Default: None, meaning no offset.
```
Comment (ngoldbaum) [inline yt/visualization/plot_modifications.py]: can you say that this uses the `time_unit` kwarg for the unit if you pass in a scalar with no explicit units?

File: yt/visualization/plot_modifications.py [hunk 2]
```diff
@@ -2000,6 +2019,12 @@ class ScaleCallback(PlotCallback):
         The image location of the scale bar in the plot coordinate system.
         Setting pos overrides the corner parameter.
 
+    format : string, optional
```
Comment (ngoldbaum) [inline yt/visualization/plot_modifications.py]: can you call this something like `scalebar_text_format`? I think `format` is a little ambiguous: format of what?. Can you also add what the default is and note that both `scale` and `format` must be specified in the string and nothing else?

File: doc/source/visualizing/callbacks.rst [hunk 3]
```diff
@@ -622,8 +622,8 @@ Add the Current Time and/or Redshift
 
 .. function:: annotate_timestamp(x_pos=None, y_pos=None, corner='lower_left',\
                                  time=True, redshift=False, \
-                                 time_format='t = {time:.0f} {units}', \
-                                 time_unit=None, \
+                                 time_format='t = {time:.1f} {units}', \
```
Comment (chummels) [inline doc/source/visualizing/callbacks.rst]: Why change to .1f?

Comment (jzuhone) [reply doc/source/visualizing/callbacks.rst]: In the current codebase it actually is `.1f`, so now the docs match the code.

File: yt/visualization/plot_modifications.py [hunk 4]
```diff
@@ -2050,9 +2078,9 @@ class ScaleCallback(PlotCallback):
     _type_name = "scale"
     _supported_geometries = ("cartesian", "spectral_cube", "force")
     def __init__(self, corner='lower_right', coeff=None, unit=None, pos=None,
-                 max_frac=0.16, min_frac=0.015, coord_system='axis',
-                 text_args=None, size_bar_args=None, draw_inset_box=False,
-                 inset_box_args=None):
+                 scale_text_format="{scale} {units}", max_frac=0.16,
```
Comment (chummels) [inline yt/visualization/plot_modifications.py]: Super minor, but maybe have scale_text_format at the end to preserve the previous kwarg ordering?

File: doc/source/visualizing/callbacks.rst (hunk 3 above)
Comment (chummels) [reply doc/source/visualizing/callbacks.rst]: Touché!  OK, sounds good!

**Pull request merged**